boards/arm/stm32/olimex-stm32-p407/scripts/memory.ld

```diff
@@ -92,7 +92,7 @@ MEMORY
 
   /* 112Kb of contiguous SRAM */
 
-  ksram (rwx)      : ORIGIN = 0x20000000, LENGTH = 8K
-  usram (rwx)      : ORIGIN = 0x20002000, LENGTH = 8K
-  xsram (rwx)      : ORIGIN = 0x20008000, LENGTH = 96K
+  ksram (rwx)      : ORIGIN = 0x20000000, LENGTH = 16K
+  usram (rwx)      : ORIGIN = 0x20004000, LENGTH = 16K
+  xsram (rwx)      : ORIGIN = 0x20008000, LENGTH = 80K
```

Comment:
why not move sigev_notify_thread_id feature to CONFIG_SIG_EVTHREAD

Reply:
According to the sigevent(3) specification, the `sigev_notify_thread_id` is a Linux-specific field and should not be confused with the `CONFIG_SIG_EVTHREAD` fields. These fields serve distinct purposes.

Reply:
Yes, although `sigev_notify_thread_id` is Linux specific, which should all be attributed by **SIGEV_THREAD**:
```
       struct sigevent {
           int    sigev_notify;  /* Notification method */
           int    sigev_signo;   /* Notification signal */
           union sigval sigev_value;
                                 /* Data passed with notification */
           void (*sigev_notify_function) (union sigval);
                                 /* Function used for thread
                                    notification (SIGEV_THREAD) */
           void  *sigev_notify_attributes;
                                 /* Attributes for notification thread
                                    (SIGEV_THREAD) */
           pid_t  sigev_notify_thread_id;
                                 /* ID of thread to signal
                                    (SIGEV_THREAD_ID); Linux-specific */
       };

```
I think here may be no need to change the RAM configuration if use `CONFIG_SIG_EVTHREAD` to control `sigev_notify_thread_id`.

Reply:
> sigev_notify_thread_id

`SIGEV_THREAD_ID` and `SIGEV_THREAD` have no functional relationship except for their similar names. They are totally independent feature.

Reply:
OK, so how about add a new config `CONFIG_SIG_EVTHREAD_ID`? It is better to avoid change the ram size of custom board especially you can't verify it.

Reply:
> OK, so how about add a new config `CONFIG_SIG_EVTHREAD_ID`? It is better to avoid change the ram size of custom board especially you can't verify it.

That's fine. I will add new config `CONFIG_SIG_EVTHREAD_ID` in next PR.

Reply:
Don't add a configure except it really need.

If you concern the size.
We have next patch for this:
typedef struct sigevent
{
  uint8_t      sigev_notify; /* Notification method: SIGEV_SIGNAL, SIGEV_NONE, or SIGEV_THREAD */
  uint8_t      sigev_signo;  /* Notification signal */
  union sigval sigev_value;  /* Data passed with notification */

  union
    {
#ifdef CONFIG_SIG_EVTHREAD
      struct
        {
          /* Notification function */

          sigev_notify_function_t _function;

          /* Notification attributes (not used) */

          FAR struct pthread_attr_s *_attribute;
        } _sigev_thread;
#endif
      pid_t _tid; /* ID of thread to signal */
    } _sigev_un;
} sigevent_t;

Reply:
if the size is concern is better to restore DISABLE_SIGNALS option.